### PR TITLE
Changed ls Sort and Styling

### LINKED ISSUE #231: Neue ls Formatierung
**Is your feature request related to a problem? Please describe.**
Eine Formatierung für `ls`.
Sortiert nach Ordner oder Datei.
Und mit Farben markiert.

**Additional Comments**
Bsp:
Neue Anzeige von Dateien/Ordnern:
[FOLDER] Ordner1     | empty folder
[FOLDER] Ordner2    | contains 2 files    | 0 folder
[FOLDER] Ordner3    | contains 5 files    | 1 folder
[FILE] Datei1.txt
[FILE] wallet.wf

+ Ordner sind blau dargestellt
+ Ordner zuerst aufgelistet
+ reguläre Dateien werden nach Ordnern aufgelistet

### PR DESCRIPTION
**Pull request by Tert0**
**Description**
Changed the Sort. of `ls`:
- Folder 
- Files

Folders are Blue.

**Issue**
Closes #231 

**Testing Instructions**
> Type in `ls` to the Terminal

### CODE REVIEW

File: src/app/desktop/windows/terminal/terminal-states.ts [hunk 1]
```diff
@@ -423,11 +424,29 @@ export class DefaultTerminalState extends CommandTerminalState {
...
+  }
+
   ls(args: string[]) {
+
+
     if (args.length === 0) {
-      this.fileService.getFiles(this.activeDevice['uuid'], this.working_dir).subscribe(files =>
-        files.forEach(f => this.terminal.outputText(f.filename))
-      );
+      this.fileService.getFiles(this.activeDevice['uuid'], this.working_dir).subscribe((files) => {
```
Comment (MarcelCoding) [inline src/app/desktop/windows/terminal/terminal-states.ts:447]:
```suggestion
      this.fileService.getFiles(this.activeDevice['uuid'], this.working_dir).subscribe(files => {
```

File: src/app/desktop/windows/terminal/terminal-states.ts [hunk 2]
```diff
@@ -423,11 +424,29 @@ export class DefaultTerminalState extends CommandTerminalState {
...
+      return !file.is_directory;
+    });
+    folders.sort().forEach(folder => {
+      this.terminal.output('<span style="color: #41ABFC;">' + folder.filename + '</span>');
+    });
+
+    files.sort().forEach(file => {
+      this.terminal.outputText(file.filename);
+    });
+  }
```
Comment (MarcelCoding) [inline src/app/desktop/windows/terminal/terminal-states.ts:441]: Why don't you chain the operators instead of using a variable?

**Review by MarcelCoding — APPROVED** [review]

**Comment by Tert0** [comment]
Please dont aprove i got some Ideas from Chaozz and Mimas

File: .vscode/settings.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   },
   "files.autoSave": "onWindowChange",
   "files.encoding": "utf8",
-  "editor.formatOnSave": true,
+  "editor.formatOnSave": false,
```
Comment (JannikEmmerich) [inline .vscode/settings.json:11]: This has noting to do with your pull request.

File: src/assets/api.json [hunk 4]
```diff
@@ -1,3 +1,4 @@
 {
-  "url": "wss://ws.test.cryptic-game.net/"
+  "url": "wss://ws.cryptic.defelo.ml",
+  "url_test": "wss://ws.cryptic-game.net"
```
Comment (JannikEmmerich) [inline src/assets/api.json:3]: This has nothing to do with your pull request.

**Review by JannikEmmerich — CHANGES_REQUESTED** [review]

**Comment by Tert0** [comment]
@JannikEmmerich Changed it.

**Comment by Tert0** [comment]
> Please dont aprove i got some Ideas from Chaozz and Mimas

This PR is now Ready

File: src/app/desktop/windows/settings/settings.component.html [hunk 5]
```diff
@@ -24,6 +24,32 @@
...
+            #lsFolderColorSetting
+            (change)="onChangeSetting()"
+            [value]="current.getSettings().lsfc"
+            type="color"
+          />
+        </label>
+      </div>
+    </div>
+    <div class="row">
+      <span>ls File/Folder Prefix</span>
```
Comment (JannikEmmerich) [inline src/app/desktop/windows/settings/settings.component.html:41]:
```suggestion
      <span>`ls` command file/folder prefix</span>
```

File: src/app/desktop/windows/settings/settings.component.html [hunk 6]
```diff
@@ -24,6 +24,32 @@
         </label>
       </div>
     </div>
+    <div class="row">
+      <span>ls Folder Color</span>
```
Comment (JannikEmmerich) [inline src/app/desktop/windows/settings/settings.component.html:28]:
```suggestion
      <span>`ls` folder color</span>
```

File: src/app/desktop/windows/settings/settings.service.ts [hunk 7]
```diff
@@ -42,4 +42,14 @@ export class SettingsService {
     return this.getSettings().tpc;
   }
 
+  // LS Folder color
```
Comment (JannikEmmerich) [inline src/app/desktop/windows/settings/settings.service.ts:45]:
```suggestion
  // ls command folder color
```

File: src/app/desktop/windows/settings/settings.service.ts [hunk 8]
```diff
@@ -42,4 +42,14 @@ export class SettingsService {
...
+  getLSFC(): string {
+    return this.getSettings().lsfc;
+  }
+
+  // LS Prefix
```
Comment (JannikEmmerich) [inline src/app/desktop/windows/settings/settings.service.ts:50]:
```suggestion
  // ls command prefix
```

File: src/app/desktop/windows/terminal/terminal-states.ts [hunk 9]
```diff
@@ -423,11 +424,28 @@ export class DefaultTerminalState extends CommandTerminalState {
...
+      return !file.is_directory;
+    }).sort().forEach(file => {
+      this.terminal.outputText(`${(this.settings.getLSPrefix() ? '[File] ' : '')}${file.filename}`);
+    });
+  }
+
   ls(args: string[]) {
+
+
     if (args.length === 0) {
```
Comment (JannikEmmerich) [inline src/app/desktop/windows/terminal/terminal-states.ts:445]:
```suggestion
  ls(args: string[]) {
    if (args.length === 0) {
```

File: src/assets/api.json [hunk 10]
```diff
@@ -1,3 +1,3 @@
 {
-  "url": "wss://ws.test.cryptic-game.net/"
+  "url": "wss://ws.test.cryptic-game.net"
```
Comment (JannikEmmerich) [inline src/assets/api.json:2]:
```suggestion
  "url": "wss://ws.test.cryptic-game.net/"
```

**Review by JannikEmmerich — CHANGES_REQUESTED** [review]

**Review by JannikEmmerich — APPROVED** [review]

**Review by MarcelCoding — APPROVED** [review]

**Pull request merged**